W-13808873-rtf-backup-restore-openshift-kt (#1082)
mulesoft/docs-runtime-fabric: modules/ROOT/pages/manage-backup-restore.adoc

The Runtime Fabric backup process affects only the Runtime Fabric cluster. The backup process does not back up data from the control plane.

> For regular Kubernetes clusters the following is backed up:

Reviewer: ```suggestion
For regular Kubernetes clusters, the following is backed up:
```

** Edge
** Anypoint Monitoring and external logforwarders

> For OpenShift clusters the following is backed up:

Reviewer: ```suggestion
For OpenShift clusters, the following is backed up:
```

----
kubectl scale --replicas=0 -n rtf deployment.apps/agent
----
+
[start =5]
+
. Restore the cluster from the backup: 
[source,copy]
----
./rtfctl restore <path_to_backup_file>
----
This process may require several minutes to complete.

For Runtime Fabric, confirm the `rtfctl` binary is present in the current directory and the Kubernetes (kubectl) context is set to the cluster you are restoring to.

== Create a Backup and Restore for OpenShift Clusters

In the following example, you have a Runtime Fabric with a set of Mule applications and configurations deployed on cluster (named `cluster-1`) inside an OpenShift cluster. To create a backup and perform a restore, follow these steps:

. Run the backup command: 
+
[source,copy]
----
./rtfctl backup <path_to_backup_file>
----
+
[start=2]
. Shutdown `cluster-1`.
. Remove and shutdown all Mule application deployments.

> . Remove Runtime Fabric installation.

Reviewer 2: ```suggestion
. Remove the Runtime Fabric installation.
```

> . Remove Runtime Fabric operator from the OpenShift cluster (optional).

Reviewer 2: ```suggestion
. Remove the Runtime Fabric operator from the OpenShift cluster (optional).
```

> . Using `curl` hit the management plane API endpoint to regenerate the activation ID. +

Reviewer 2: ```suggestion
. Using `curl`, hit the management plane API endpoint to regenerate the activation ID. +
```

Reviewer 2: I'm not sure "hit" will translate well here, although it makes sense in English. Is there another word we can use? "Access"? I don't know if that changes meaning...

> The Runtime Fabric will show as ready for activation so you can install and activate Runtime Fabric again on a new OpenShift cluster. 

Reviewer 2: ```suggestion
The Runtime Fabric will show as ready for activation. You can install and activate Runtime Fabric again on a new OpenShift cluster. 
```

+
[source,copy]
----
curl --location --request PATCH 'https://anypoint.mulesoft.com/runtimefabric/api/organizations/{org-id}/fabrics/{fabric-id}/regenerateActivationData' \
--header 'Authorization: Bearer {bearer-token}
----
+
[start=7]
. Install the Runtime Fabric operator on `cluster-2` using the regenerated activation data ID. +
The installation must be on the same Runtime Fabric namespace because all labels on the backup point to that installation.
. Run the restore command to pick the set of resources and restore them in the cluster:
+
[source,copy]
----
./rtfctl restore <path_to_backup_file>
----

> +

Reviewer 3: I would add here a simple step stating that after restoring you should check on the status of runtime fabric/deployments.

----
kubectl scale --replicas=0 -n rtf deployment.apps/agent
----
+
[start =5]
+
. Restore the cluster from the backup: 
[source,copy]
----
./rtfctl restore <path_to_backup_file>
----
This process may require several minutes to complete.

For Runtime Fabric, confirm the `rtfctl` binary is present in the current directory and the Kubernetes (kubectl) context is set to the cluster you are restoring to.

== Create a Backup and Restore for OpenShift Clusters

In the following example, you have a Runtime Fabric with a set of Mule applications and configurations deployed on cluster (named `cluster-1`) inside an OpenShift cluster. To create a backup and perform a restore, follow these steps:

. Run the backup command: 
+
[source,copy]
----

> ./rtfctl backup <path_to_backup_file>

Reviewer 4: ./rtfctl backup <path_to_backup_file> -n <rtf_namespace>

----
+
[start=2]
. Shutdown `cluster-1`.

> . Remove and shutdown all Mule application deployments.

Reviewer 4: Run kubectl command to delete Mule applications.  Don't remove applications from Runtime Manager in UI.
ie.  `kubectl delete ns <app_namespace>`

. Remove the Runtime Fabric installation.
. Remove the Runtime Fabric operator from the OpenShift cluster (optional).
. Send a `curl` command to the management plane API endpoint to regenerate the activation ID. +
The Runtime Fabric will show as ready for activation. You can install and activate Runtime Fabric again on a new OpenShift cluster. 
+
[source,copy]
----
curl --location --request PATCH 'https://anypoint.mulesoft.com/runtimefabric/api/organizations/{org-id}/fabrics/{fabric-id}/regenerateActivationData' \
--header 'Authorization: Bearer {bearer-token}
----
+
[start=7]
. Install the Runtime Fabric operator on `cluster-2` using the regenerated activation data ID. +
The installation must be on the same Runtime Fabric namespace because all labels on the backup point to that installation.
. Run the restore command to pick the set of resources and restore them in the cluster:
+
[source,copy]
----

> ./rtfctl restore <path_to_backup_file>

Reviewer 4: ./rtfctl restore <path_to_backup_file> -n <rtf_namespace>

In the following example, you have a Runtime Fabric with a set of Mule applications and configurations deployed on cluster (named `cluster-1`) inside an OpenShift cluster. To create a backup and perform a restore, follow these steps:

. Run the backup command: 
+
[source,copy]
----
./rtfctl backup <path_to_backup_file> -n <rtf_namespace>
----
+
[NOTE]
MuleSoft performs the back up based on the detected cluster you have. Though, you can also manually specify the type of back up to perform using the flag `--backup-type` string. The flag options are `auto`, `openshift`, and `full`. With the default `auto` flag option, MuleSoft automatically detects the infrastructure cluster provider to perform the back up.
[start=2]
. Shutdown `cluster-1`.
. Delete all Mule application deployments by running the following command:
+
[source,copy]
----
kubectl delete ns <app_namespace>
----
+
Do not remove applications from the Runtime Manager UI.
[start=4]
. Remove the Runtime Fabric installation.
. Remove the Runtime Fabric operator from the OpenShift cluster (optional).
. Send a `curl` command to the management plane API endpoint to regenerate the activation ID. +

> The Runtime Fabric will show as ready for activation. You can install and activate Runtime Fabric again on a new OpenShift cluster. 

Reviewer: ```suggestion
The Runtime Fabric shows as ready for activation. You can install and activate Runtime Fabric again on a new OpenShift cluster. 
```

+
[source,copy]
----
curl --location --request PATCH 'https://anypoint.mulesoft.com/runtimefabric/api/organizations/{org-id}/fabrics/{fabric-id}/regenerateActivationData' \
--header 'Authorization: Bearer {bearer-token}
----
+
[start=7]
. Install the Runtime Fabric operator on `cluster-2` using the regenerated activation data ID. +

> The installation must be on the same Runtime Fabric namespace because all labels on the backup point to that installation.

Reviewer: ```suggestion
The installation must be on the same Runtime Fabric namespace because all labels on the backup point to that installation.
```

If you use `because`, you're missing a part of the sentence.
